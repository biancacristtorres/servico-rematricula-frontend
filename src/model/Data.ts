export class Data {
  public data!: Date;
  public horaInicio!: Date;
  public horaFim!: Date;
}
