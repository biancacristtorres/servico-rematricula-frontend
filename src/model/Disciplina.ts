import { SituacaoDaDisciplina } from '@/common/enum/SituacaoDaDisciplina';
import { Turma } from './Turma';

export class Disciplina {
  public codigo: number;
  public nome: string;
  public situacao: SituacaoDaDisciplina;
  public modulo: string;
  public eDisciplinaEletiva: boolean;
  public possuiEquivalenciaDoisPorUm!: boolean;
  public turmas: Turma[] = [];
  public turmasEquivalentes: Turma[] = [];
  public turmasEspeciais: Turma[] = [];

  constructor(codigo: number, nome: string, situacao: SituacaoDaDisciplina,
              modulo: string, eDisciplinaEletiva: boolean) {
    this.codigo = codigo;
    this.nome = nome;
    this.situacao = situacao;
    this.modulo = modulo;
    this.eDisciplinaEletiva = eDisciplinaEletiva;
  }
}
