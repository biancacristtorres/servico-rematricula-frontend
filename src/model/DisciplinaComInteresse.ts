export class DisciplinaComInteresse {
  public codigo?: number;
  public nome?: string;
  public interesseManifestado?: boolean;

  constructor(codigo: number, nome: string, interesseManifestado: boolean) {
    this.codigo = codigo;
    this.nome = nome;
    this.interesseManifestado = interesseManifestado;
  }
}
