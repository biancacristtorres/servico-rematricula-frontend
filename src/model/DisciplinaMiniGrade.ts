export class DisciplinaMiniGrade {
  public codigoDisciplina?: number;
  public nomeDisciplina?: string;
}
