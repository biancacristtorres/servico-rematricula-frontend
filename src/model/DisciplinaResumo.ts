import { StatusDaDisciplina } from '@/common/enum/StatusDaDisciplina';

export class DisciplinaResumo {
  public codigoDisciplina?: number;
  public nomeDisciplina?: string;
  public descricaoTurma?: string;
  public status?: StatusDaDisciplina;
  public ePorData?: boolean;
  public campus?: string;
  public descricaoTipoCargaHoraria?: string;
  public datas?: string[];
  public nomeDisciplinaEquivalente?: string;
}
