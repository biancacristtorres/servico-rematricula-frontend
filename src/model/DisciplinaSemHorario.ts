export class DisciplinaSemHorario {
  public nomeDisciplina?: string;
  public nomeDisciplinaEquivalente?: string;
}
