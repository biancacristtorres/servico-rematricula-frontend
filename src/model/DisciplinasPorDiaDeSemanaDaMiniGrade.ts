import { DisciplinaMiniGrade } from './DisciplinaMiniGrade';

export class DisciplinasPorDiaDeSemanaDaMiniGrade {
  [index: number]: DisciplinaMiniGrade[];
  public 2: DisciplinaMiniGrade[];
  public 3: DisciplinaMiniGrade[];
  public 4: DisciplinaMiniGrade[];
  public 5: DisciplinaMiniGrade[];
  public 6: DisciplinaMiniGrade[];
  public 7: DisciplinaMiniGrade[];
}
