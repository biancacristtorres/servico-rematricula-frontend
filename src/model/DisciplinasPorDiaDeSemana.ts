import { DisciplinaResumo } from './DisciplinaResumo';

export class DisciplinasPorDiaDeSemana {
  [index: number]: DisciplinaResumo[];
  public 2: DisciplinaResumo[];
  public 3: DisciplinaResumo[];
  public 4: DisciplinaResumo[];
  public 5: DisciplinaResumo[];
  public 6: DisciplinaResumo[];
  public 7: DisciplinaResumo[];
}
