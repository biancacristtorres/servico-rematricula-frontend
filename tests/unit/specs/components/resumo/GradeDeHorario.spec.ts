import {
  GradeDeHorarioFactory,
} from '../../../factory/components/resumo/GradeDeHorarioFactory';
import { DisciplinaResumo } from '@/model/DisciplinaResumo';

describe('Componente de GradeDeHorario', () => {
  const factory = new GradeDeHorarioFactory();

  beforeEach(() => {
    factory.prepararContainerDeInjecaoDaApi();
    factory.dadoQueAApiDeResumoRetorneDisciplinaComTurmaEspecialComSucesso();
  });

  afterEach(() => {
    factory.restaurarContainerDeInjecaoDaApi();
  });

  it('snapshot', async (done) => {
    await factory.montarComponente();

    expect(factory.componente)
      .toMatchSnapshot();

    done();
  });

  it('Exibe ícone de turma especial', async (done) => {
    await factory.montarComponente();

    expect(factory.getPossuiDisciplinaTurmaEspecial).toBeTruthy();

    done();
  });

  it('Quando uma disciplina não possui equivalente ' +
    'deve exibir somente o nome de uma disciplina', async (done) => {
    const disciplina = new DisciplinaResumo();
    disciplina.nomeDisciplina = 'Disciplina Da Grade';
    disciplina.descricaoTurma = 'XXXX-XXX';

    await factory.montarComponente();
    const atual = factory.vueInstance.formatarNomeDisciplina(disciplina);

    const esperado = 'Disciplina Da Grade (XXXX-XXX)';
    expect(atual).toEqual(esperado);

    done();
  });

  it('Quando uma disciplina possui uma equivalente com nome diferente da grade ' +
    'deve exibir o nome de ambas', async (done) => {
    const disciplina = new DisciplinaResumo();
    disciplina.nomeDisciplina = 'Disciplina Da Grade';
    disciplina.nomeDisciplinaEquivalente = 'Disciplina Equivalente';
    disciplina.descricaoTurma = 'XXXX-XXX';

    await factory.montarComponente();
    const atual = factory.vueInstance.formatarNomeDisciplina(disciplina);

    const esperado = 'Disciplina Equivalente (Eq. à Disciplina Da Grade)<br/> (XXXX-XXX)';
    expect(atual).toEqual(esperado);

    done();
  });

  it('Quando uma disciplina possui uma equivalente com nome igual ao da grade ' +
    'deve exibir somente o nome de uma disciplina', async (done) => {
    const disciplina = new DisciplinaResumo();
    disciplina.nomeDisciplina = 'Disciplina Da Grade';
    disciplina.nomeDisciplinaEquivalente = disciplina.nomeDisciplina;
    disciplina.descricaoTurma = 'XXXX-XXX';

    await factory.montarComponente();
    const atual = factory.vueInstance.formatarNomeDisciplina(disciplina);

    const esperado = 'Disciplina Da Grade (XXXX-XXX)';
    expect(atual).toEqual(esperado);

    done();
  });
});
