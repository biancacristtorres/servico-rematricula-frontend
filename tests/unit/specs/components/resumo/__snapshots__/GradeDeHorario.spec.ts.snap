// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de GradeDeHorario snapshot 1`] = `
<div class="pr-2">
  <div class="container-tabela">
    <div class="container box pa-0 caption">
      <div class="layout row">
        <div class="flex display-table xs2">
          <div class="celula celula-cabecalho display-cell font-weight-bold">Horários</div>
        </div>
        <div class="flex display-table xs2">
          <div class="celula celula-cabecalho display-cell font-weight-bold">Segunda</div>
        </div>
        <div class="flex display-table xs2">
          <div class="celula celula-cabecalho display-cell font-weight-bold">Terça</div>
        </div>
        <div class="flex display-table xs2">
          <div class="celula celula-cabecalho display-cell font-weight-bold">Quarta</div>
        </div>
        <div class="flex display-table xs2">
          <div class="celula celula-cabecalho display-cell font-weight-bold">Quinta</div>
        </div>
        <div class="flex display-table xs2">
          <div class="celula celula-cabecalho display-cell font-weight-bold">Sexta</div>
        </div>
        <div class="flex display-table xs2">
          <div class="celula celula-cabecalho display-cell font-weight-bold">Sábado</div>
        </div>
      </div>
      <!---->
      <div class="layout row">
        <div class="flex display-table xs2">
          <div class="celula pt-2 display-cell font-weight-bold">19:00 - 19:50h</div>
        </div>
        <div class="flex xs2">
          <div class="container pa-0 celula fill-height">
            <div class="layout row wrap"></div>
          </div>
        </div>
        <div class="flex xs2">
          <div class="container pa-0 celula fill-height">
            <div class="layout row wrap">
              <div class="flex display-table xs12"><img style="width: 0px; height: 0px; position: relative; border-top: 16px solid #082e6d; border-right: 16px solid transparent;">
                <div class="display-cell pa-2">
                  <!---->
                  <div class="celula-disciplina">
                    <!---->
                    <!---->
                    <div><i aria-hidden="true" class="v-icon icone-turma-especial mdi mdi-book-open-page-variant theme--light grey--text text--darken-1" style="font-size: 16px;"></i></div>
                  </div>
                  <div class="display-table" style="table-layout: fixed; width: 100%;">
                    <div class="display-cell">
                      <p class="celula-disciplina">Laboratório (XXXXX)</p>
                    </div>
                  </div>
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="flex xs2">
          <div class="container pa-0 celula fill-height">
            <div class="layout row wrap"></div>
          </div>
        </div>
        <div class="flex xs2">
          <div class="container pa-0 celula fill-height">
            <div class="layout row wrap"></div>
          </div>
        </div>
        <div class="flex xs2">
          <div class="container pa-0 celula fill-height">
            <div class="layout row wrap"></div>
          </div>
        </div>
        <div class="flex xs2">
          <div class="container pa-0 celula fill-height">
            <div class="layout row wrap"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <p class="text-xs-left pa-0 pt-2"><span class="body-2">Legenda:</span> <span class="legenda text-xs-left caption pl-3"><i aria-hidden="true" class="v-icon pr-1 material-icons theme--light green--text" style="font-size: 16px;">check</i>Disciplina Confirmada
    </span> <span class="legenda text-xs-left caption pl-3"><i aria-hidden="true" class="v-icon pr-1 material-icons theme--light orange--text" style="font-size: 16px;">repeat</i>Aguardando Confirmação
    </span> <span class="legenda text-xs-left caption pl-3"><i aria-hidden="true" class="v-icon pr-1 material-icons theme--light orange--text" style="font-size: 16px;">restore</i>Fila de Espera
    </span> <span class="legenda text-xs-left caption pl-3"><i aria-hidden="true" class="v-icon pr-1 mdi mdi-book-open-page-variant theme--light grey--text text--darken-1" style="font-size: 16px;"></i>Turma Especial
    </span></p>
</div>
`;
